modules/core/src/lib/layer.js

```diff
@@ -159,6 +160,11 @@ export default class Layer extends Component {
     return this.internalState && this.internalState.layer;
   }
 
+  // Returns the default parse options for async props
+  getLoadOptions() {
```

Comment:
Makes sense, but can we make `_getLoadOptions` underscore for now? I do not yet have a good feeling about this part of the design, it is one of the things I wanted to address in the RFC.

A minor objection is that with these changes, the `scenegraph` load options will also apply to `data` loads (could be addressed by passing in the prop name).

More importantly, I have been struggling quite a bit with the `ScenegraphLayer`, trying to load things on the application side and passing them in, but not being able to do so because of things like `getLoadOptions()` not being available outside of the layer.

I note that unfortunately some options (like `gl`) are not static and must be provided by the layer, however others could be declaratively specified, perhaps as part of the default props definition?

Also we probably want a way for the layer user to override or replace loader options (perhaps per prop or per loader).

loaders.gl is intended to support an array syntax like `[CSVLoader, {header: true}]` to specify a loader with overridden default options. So if we can supply a list of loader additions/overrides to layers that should offer a decent amount of flexibility to the end user.

Reply:
It might make sense to give the options object some sort of structure? Right now it is being passed to `fetch` as well as any loader that is automatically picked.

What if we support something like 

```js
load(url, {
  loaders: [...],
  fetch: {...},
  json: {...},
  csv: {...}
})
```

Reply:
> What if we support something like

Yes that structure makes perfect sense (the loaders.gl options object is currently excessively overloaded and hard to document and use) and as an API it also looks very pretty.

Now, to make this consistent there are a bunch of little details/complications that need to be "sweated" (discussed, decided upon and implemented).

I've opened a suite of small RFCs on loaders https://github.com/uber-web/loaders.gl/pull/269 to help us progress on that, and done a pass on the deck.gl loaders.gl RFC to reflect the changes represented by this PR.